flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/maintenance/operator/AntiJoin.java

```diff
@@ -0,0 +1,133 @@
+import org.apache.flink.api.common.state.ValueState;
+import org.apache.flink.api.common.state.ValueStateDescriptor;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.streaming.api.functions.co.KeyedCoProcessFunction;
+import org.apache.flink.util.Collector;
+import org.apache.iceberg.actions.DeleteOrphanFiles;
+import org.apache.iceberg.actions.FileURI;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Internal
+public class AntiJoin extends KeyedCoProcessFunction<String, String, String, String> {
+  private static final Logger LOG = LoggerFactory.getLogger(AntiJoin.class);
+
+  private transient ListState<String> foundInTable;
+  private transient ValueState<String> foundInFileSystem;
+  private final DeleteOrphanFiles.PrefixMismatchMode prefixMismatchMode;
+  private final Map<String, String> equalSchemes;
+  private final Map<String, String> equalAuthorities;
+
+  public AntiJoin(
+      DeleteOrphanFiles.PrefixMismatchMode prefixMismatchMode,
+      Map<String, String> equalSchemes,
+      Map<String, String> equalAuthorities) {
+    this.prefixMismatchMode = prefixMismatchMode;
+    this.equalSchemes = equalSchemes;
+    this.equalAuthorities = equalAuthorities;
+  }
+
+  @Override
+  public void open(OpenContext openContext) throws Exception {
+    super.open(openContext);
+    foundInTable =
+        getRuntimeContext()
+            .getListState(new ListStateDescriptor<>("antiJoinFoundInTable", Types.STRING));
+    foundInFileSystem =
+        getRuntimeContext()
+            .getState(new ValueStateDescriptor<>("antiJoinFoundInFileSystem", Types.STRING));
+  }
+
+  @Override
+  public void processElement1(String value, Context context, Collector<String> collector)
+      throws Exception {
+    foundInTable.add(value);
+    context.timerService().registerEventTimeTimer(context.timestamp());
+  }
+
+  @Override
+  public void processElement2(String value, Context context, Collector<String> collector)
+      throws Exception {
+    foundInFileSystem.update(value);
+    context.timerService().registerEventTimeTimer(context.timestamp());
+  }
+
+  @Override
+  public void onTimer(long timestamp, OnTimerContext ctx, Collector<String> out) throws Exception {
+    List<FileURI> foundInTablesList = Lists.newArrayList();
+    for (String uri : foundInTable.get()) {
+      foundInTablesList.add(new FileURI(uri, equalSchemes, equalAuthorities));
+    }
+
+    if (foundInFileSystem.value() != null && foundInTablesList.isEmpty()) {
+      FileURI fileURI = new FileURI(foundInFileSystem.value(), equalSchemes, equalAuthorities);
+      out.collect(fileURI.getUriAsString());
+    } else if (foundInFileSystem.value() != null && !foundInTablesList.isEmpty()) {
+      FileURI actual = new FileURI(foundInFileSystem.value(), equalSchemes, equalAuthorities);
+      boolean found = true;
```

Comment:
the naming seems off.
What does `found` mean here? `foundAnError`?
Maybe it is easier to understand if we start `boolean mismatch=false`